### PR TITLE
fix: properly seriailize meta buffer in SendStringArrInternal

### LINKED ISSUE #3449: Regression: long replies contain invalid characters
How to reproduce:

```python
@pytest.mark.asyncio
async def test_large_replies(df_factory):
    master = df_factory.create(proactor_threads=1)
    df_factory.start_all([master])
    c_master = master.client()

    l = [i for i in range(0, 10_000)]
    await c_master.lpush("list", *l)
    assert l == await c_master.lrange("list", 0, -1)

    await c_master.close()
```

Error:

```
$ redis-cli lrange _counter1 0 -1
Error: Protocol error, got "\r" as reply type byte
```

### PR DESCRIPTION
**Pull request by romange**
Fixes #3449 that was introduced by #3425

### CODE REVIEW

File: src/facade/reply_builder.cc [hunk 1]
```diff
@@ -593,47 +593,54 @@ void RedisReplyBuilder::SendStringArrInternal(
   serialize_len(type_char[0], header_len);
   unsigned vec_indx = 0;
   string_view src;
+
+#define FLUSH_IOVEC()           \
```
Comment (kostasrim) [inline src/facade/reply_builder.cc:597]: Why is this a macro and not a lambda?

Comment (romange) [reply src/facade/reply_builder.cc:597]: because it returns from the parent scope

Comment (kostasrim) [reply src/facade/reply_builder.cc:597]: sure and in that case you just write:

```
      if (vec_indx >= vec_cap && FLUSH_IOVEC()) {
        return;
      }
```

Which IMO is almost the same. It's a nit, don't pay much attention to it if you don't want to :)

**Review by kostasrim — APPROVED** [review]

**Comment by kostasrim** [comment]
@romange tiered_storage memory pressure failed:

```
2024-08-06T06:55:27.8418416Z 49: ../src/server/tiered_storage_test.cc:311: Failure
2024-08-06T06:55:27.8419232Z 49: Value of: true
2024-08-06T06:55:27.8420000Z 49:   Actual: true
2024-08-06T06:55:27.8420518Z 49: Expected: false
```

**Comment by kostasrim** [comment]
> @romange tiered_storage memory pressure failed:
> 
> ```
> 2024-08-06T06:55:27.8418416Z 49: ../src/server/tiered_storage_test.cc:311: Failure
> 2024-08-06T06:55:27.8419232Z 49: Value of: true
> 2024-08-06T06:55:27.8420000Z 49:   Actual: true
> 2024-08-06T06:55:27.8420518Z 49: Expected: false
> ```

seems irrelevant I am rerunning the tests and I will also fill an issue

**Comment by kostasrim** [comment]
> > @romange tiered_storage memory pressure failed:
> > ```
> > 2024-08-06T06:55:27.8418416Z 49: ../src/server/tiered_storage_test.cc:311: Failure
> > 2024-08-06T06:55:27.8419232Z 49: Value of: true
> > 2024-08-06T06:55:27.8420000Z 49:   Actual: true
> > 2024-08-06T06:55:27.8420518Z 49: Expected: false
> > ```
> 
> seems irrelevant I am rerunning the tests and I will also fill an issue

filled

https://github.com/dragonflydb/dragonfly/issues/3456

**Pull request merged**